django/api/services/tests/test_calculate_rebate.py

```diff
@@ -0,0 +1,177 @@
+from unittest import TestCase
+from rest_framework.serializers import ValidationError
+from django.test import TestCase
+from ...models.go_electric_rebate_application import GoElectricRebateApplication
+from ...models.household_member import HouseholdMember
+from django.contrib.auth import get_user_model
+from ..calculate_rebate import calculate_rebate_amount
+
+
+class TestCalculate(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        user = get_user_model()
+        user.objects.create(id=1, username="tester")
+        user.objects.create(id=2, username="tester2")
+
+        GoElectricRebateApplication.objects.create(
+            id="B5t92XeH7NnFUwxc",
+            user_id=1,
+            sin="302435839",
+            application_type="individual",
+            last_name="person",
+            first_name="person",
+            doc1="image1.png",
+            doc2="image2.png",
+            consent_personal=True,
+            consent_tax=True,
+            date_of_birth="2000-01-01",
+            email="emily.hillier@gov.bc.ca",
+            address="123 street",
+            city="Victoria",
+            postal_code="v8s4j9",
+            tax_year=2020,
+        )
+        HouseholdMember.objects.create(
+            application_id="B5t92XeH7NnFUwxc",
+            user_id=2,
+            sin="270300379",
+            last_name="test",
+            first_name="test",
+            doc1="image1.png",
+            doc2="image2.png",
+            consent_personal=True,
+            consent_tax=True,
+            date_of_birth="2000-01-01",
+        )
+
+    def test_application_a_ind(self):
+        # qualifies for an individual rebate of a
+        cra_response = {
+            "B5t92XeH7NnFUwxc": [
+                {"sin": "302435839", "year": "2020", "income": "65687"}
+            ],
+        }
+
+        individual = GoElectricRebateApplication.objects.get(id="B5t92XeH7NnFUwxc")
+        rebate_amount = calculate_rebate_amount(cra_response, individual.id)
+        self.assertEqual(rebate_amount, 4000)
+
+    def test_application_b_ind(self):
```

Comment:
These test function names are usually thought of as "sentences" so that when one fails, someone can easily read this as a sentence (assuming underscores become spaces) to find out what is going wrong.

Reply:
Yes it's a bit cryptic

`test_individual_receives_rebate_b` something like this, where the test can be interpreted by someone who didn't write this test when it fails.